packages/cli/src/cmds/preview/preview.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+"use strict";
+
+import * as path from "path";
+import { Argv } from "yargs";
+import { YargsCommand } from "../../yargsCommand";
+import { err, FxError, ok, Result } from "@microsoft/teamsfx-api";
+
+import * as utils from "../../utils";
+import * as commonUtils from "./commonUtils";
+import * as constants from "./constants";
+import { CliTelemetry } from "../../telemetry/cliTelemetry";
+import { ExclusiveLocalRemoteOptions, WorkspaceNotSupported } from "./errors";
+
+export default class Preview extends YargsCommand {
+    public readonly commandHead = `preview`;
+    public readonly command = `${this.commandHead}`;
+    public readonly description = "Preview the current application.";
+
+    public builder(yargs: Argv): Argv<any> {
+        yargs.option("local", {
+            description: "Preview the application from local, exclusive with --remote",
+            boolean: true,
+            default: false,
+        });
+        yargs.option("remote", {
+            description: "Preview the application from remote, exclusive with --local",
+            boolean: true,
+            default: false,
+        });
+        yargs.option("folder", {
+            description: "Select root folder of the project",
+            string: true,
+            default: "./",
+        });
+
+        return yargs.version(false);
+    }
+
+    public async runCommand(args: { [argName: string]: boolean | string | string[] | undefined }): Promise<Result<null, FxError>> {
+        if (args.local && args.remote) {
```

Comment:
Should you send out telemetries for preview command?

Reply:
Nice reminder, thanks. All telemetries will be added after the implementation in one single PR. Will let you know then.

Reply:
I loop you in this PR mainly to review the preview command (see the picture above).